### PR TITLE
Attribute parenthesis completion

### PR DESCRIPTION
**Pull request by nicholastmosher**
This PR implements issue #2715 for completing parentheses typed inside of attributes (`#[]` and `#![]`).

Please let me know if there are any edge cases I may have missed or extra tests that I should add! :)

@Undin

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/typing/RsMetaTagParenthesisHandler.kt [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+import com.intellij.openapi.editor.EditorModificationUtil
+import com.intellij.openapi.editor.ex.EditorEx
+import com.intellij.openapi.editor.highlighter.HighlighterIterator
+import com.intellij.openapi.fileTypes.FileType
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiFile
+import org.rust.lang.core.psi.RsElementTypes.*
+import org.rust.lang.core.psi.RsFile
+
+class RsMetaTagParenthesisHandler : TypedHandlerDelegate() {
```
Comment (Undin) [inline src/main/kotlin/org/rust/ide/typing/RsMetaTagParenthesisHandler.kt]: Let's rename it to `RsMetaItemParenthesisTypedHandler` because [we](https://github.com/intellij-rust/intellij-rust/blob/master/src/main/grammars/RustParser.bnf#L129-L135) and [rust](https://doc.rust-lang.org/reference/attributes.html) use `MetaItem` as a name of ast nodes inside attributes. Or even to `RsParenthesisTypedHandler` to avoid renaming when we need to add parenthesis completion in some other place in future

**Comment by Undin** [comment]
@nicholastmosher Note, I will have a vacation in the next few days so there will be some delay with answers/code review

**Comment by nicholastmosher** [comment]
@Undin thanks for the feedback! I'll look at reorganizing some of this and hopefully make an update in the next week or so.

**Comment by nicholastmosher** [comment]
@Undin I finally got around to refactoring the completion code. Please let me know if there's any other changes you'd like to see :)

**Comment by Undin** [comment]
@nicholastmosher Could you squash commits?

**Comment by nicholastmosher** [comment]
@Undin done!

**Comment by Undin** [comment]
Thanks!
Welcome to IntelliJ Rust contributors!
bors r+
